### PR TITLE
[Experimental ]Save tensor and keras step times initial impl

### PR DESCRIPTION
**Pull request by Vikas-kum**
### Description of changes:
save tensor inital impl
Keras logging step times.

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by vandanavk** [comment]
timeline writer output currently looks like this

```
[{"name": "Step time", "ph": "X", "ts": 1589243455481154, "pid": "worker_0", "dur": 61339, "args": {"step number": 0}},
{"name": "Step time", "ph": "X", "ts": 1589243455722920, "pid": "worker_0", "dur": 2556, "args": {"step number": 1}},
{"name": "Step time", "ph": "X", "ts": 1589243455722957, "pid": "worker_0", "dur": 2859, "args": {"step number": 2}},
{"name": "Step time", "ph": "X", "ts": 1589243455723002, "pid": "worker_0", "dur": 2503, "args": {"step number": 3}},
{"name": "Step time", "ph": "X", "ts": 1589243455723046, "pid": "worker_0", "dur": 2978, "args": {"step number": 4}},
{"name": "Step time", "ph": "X", "ts": 1589243455723087, "pid": "worker_0", "dur": 2537, "args": {"step number": 5}},
{"name": "Step time", "ph": "X", "ts": 1589243455723119, "pid": "worker_0", "dur": 2900, "args": {"step number": 6}},
{"name": "Step time", "ph": "X", "ts": 1589243455723151, "pid": "worker_0", "dur": 2277, "args": {"step number": 7}},
{"name": "Step time", "ph": "X", "ts": 1589243455723180, "pid": "worker_0", "dur": 2361, "args": {"step number": 8}},
{"name": "Step time", "ph": "X", "ts": 1589243455723209, "pid": "worker_0", "dur": 2549, "args": {"step number": 9}},
{"name": "Step time", "ph": "X", "ts": 1589243456763559, "pid": "worker_0", "dur": 7952, "args": {"step number": 24}},
{"name": "Step time", "ph": "X", "ts": 1589243456763590, "pid": "worker_0", "dur": 3695, "args": {"step number": 25}},
{"name": "Step time", "ph": "X", "ts": 1589243456763629, "pid": "worker_0", "dur": 4541, "args": {"step number": 26}},
{"name": "Step time", "ph": "X", "ts": 1589243456763660, "pid": "worker_0", "dur": 3824, "args": {"step number": 27}},
{"name": "Step time", "ph": "X", "ts": 1589243456763691, "pid": "worker_0", "dur": 4084, "args": {"step number": 28}},
{"name": "Step time", "ph": "X", "ts": 1589243456763748, "pid": "worker_0", "dur": 4191, "args": {"step number": 29}},
{"name": "Step time", "ph": "X", "ts": 1589243456763797, "pid": "worker_0", "dur": 3695, "args": {"step number": 30}},
{"name": "Step time", "ph": "X", "ts": 1589243456763832, "pid": "worker_0", "dur": 4728, "args": {"step number": 31}},
{"name": "Step time", "ph": "X", "ts": 1589243456763865, "pid": "worker_0", "dur": 3976, "args": {"step number": 32}},
{"name": "Step time", "ph": "X", "ts": 1589243456763895, "pid": "worker_0", "dur": 4081, "args": {"step number": 33}},

```

Marked some TODOs to complete

File: smdebug/core/hook.py [hunk 1]
```diff
@@ -222,6 +225,12 @@ def __init__(
         self.mode_steps = {ModeKeys.GLOBAL: init_step}
         self.writer = None
 
+        # TODO: kannanva: Don't access env var here. Make timeline writer implementation
+        # similar to tb writer.
+        self.timeline_writer = TimelineWriter(
+            os.getenv(SM_PROFILER_FILE_PATH_ENV_STR, DEFAULT_SAGEMAKER_PROFILER_PATH)
+        )
```
Comment (vandanavk) [inline smdebug/core/hook.py:232]: probably could create TimelineWriter within FileWriter like the rest of the writers. But timelinewriter doesn't use the functions of FileWriter

File: smdebug/core/hook.py [hunk 2]
```diff
@@ -477,6 +490,23 @@ def _maybe_get_tb_writer(self) -> Optional[FileWriter]:
             )
             return self.tb_writers[self.mode]
 
+    def _maybe_get_timeline_writer(self) -> Optional[FileWriter]:
+        """ Returns a FileWriter object if `hook.tensorboard_dir` has been specified, else None.
```
Comment (Vikas-kum) [inline smdebug/core/hook.py:494]: comment?

File: smdebug/core/hook.py [hunk 3]
```diff
@@ -477,6 +490,23 @@ def _maybe_get_tb_writer(self) -> Optional[FileWriter]:
...
+
+        Creates a writer if does not exist.
+        """
+        if not self.timeline_writer:
+            # s = self.step
```
Comment (Vikas-kum) [inline smdebug/core/hook.py:499]: remove

**Pull request closed without merging**